tensorflow_addons/layers/crf.py

```diff
@@ -0,0 +1,474 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Orginal implementation from keras_contrib/layers/crf
+# ==============================================================================
+"""Implementing Conditional Random Field layer."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.text.crf import crf_decode, crf_log_likelihood
+from tensorflow_addons.utils import keras_utils
+
+
+@keras_utils.register_keras_custom_object
+class CRF(tf.keras.layers.Layer):
+    """Linear chain conditional random field (CRF).
+
+    Examples:
+
+    ```python
+        from tensorflow_addons.layers import CRF
+
+        model = Sequential()
+        model.add(Embedding(3001, 300, mask_zero=True)
+
+        crf = CRF(10, name='crf_layer')
+        model.add(crf)
+
+        model.compile('adam', loss={'crf_layer': crf.loss})
```

Comment:
maybe I'm wrong, but what will the crf.loss get? The traceback says 'CRF' object has no attribute 'loss'.

Reply:
I built a model with this yesterday, and the documentation is indeed wrong ATM.

`crf.loss` would need to be replaced with `ConditionalRandomFieldLoss()` from the `losses` module to get it working

Reply:
> I built a model with this yesterday, and the documentation is indeed wrong ATM.
> 
> `crf.loss` would need to be replaced with `ConditionalRandomFieldLoss()` from the `losses` module to get it working

Thanks for your reply. So what we should do is to replace crf.loss with ConditionalRandomFieldLoss(),and the y of model.fit should be a [batch_size,sequence_length] array, is that right? The loss of my model stays around 16-20, which is abnormal, and in contrast with the result I got from keras+keras-contrib.

Reply:
Yeah, that sounds about right.

I think it's very hard to compare this to the CRF from keras-contrib, as they seem to be quite different.

Reply:
> Yeah, that sounds about right.
> 
> I think it's very hard to compare this to the CRF from keras-contrib, as they seem to be quite different.

Thank you!